### PR TITLE
Blaze: Add loading and error states for target device and language picker views

### LINKED ISSUE #11655: Loading state & error state for picker screens
We should show loading indicator for the loading state and add error handling for target option picker screens.

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #11655 

## Description

This PR refactors the `MultiSelectionView` to not include `NavigationView` and move the navigation setup to the picker views for Blaze target options. This is to make it flexible to add states to the views.

Changes:
- Injected selected items to the view model of the target option views instead of the views.
- Added states for the target option views.
- Disabled the Save button if the state is syncing/error or if the selected items are empty.
- Moved the navigation view to the target option views and removed it from the `MultiSelectionView`.

## Testing instructions

- Log in to a store with at least one existing product and is eligible for Blaze.
- Navigate to the Products tab and select any public product.
- On the product form, select Promote with Blaze.
- Notice that the campaign creation form is presented. Select the device field.
- Confirm the following:
  - Selection works properly. 
  - The device field on the creation form is updated following the selected items.
  - The Save button is enabled when there is data on the screen, and any item is selected.
  -  Hardcode the view's state - confirm that the UI is updated correctly.
- Repeat the test for the target language screen

## Screenshots

Loading state | Error state
----- | ----- 
 | 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Blaze/TargetDevices/BlazeTargetDevicePickerView.swift [hunk 1]
```diff
@@ -6,52 +6,95 @@ struct BlazeTargetDevicePickerView: View {
...
+            comment: "Button to dismiss the target device picker for campaign creation"
+        )
+        static let saveButtonTitle = NSLocalizedString(
+            "blazeTargetDevicePickerView.save",
+            value: "Save",
+            comment: "Button to save the selections on the target device picker for campaign creation"
+        )
+        static let errorMessage = NSLocalizedString(
+            "blazeTargetDevicePickerView.errorMessage",
+            value: "Error syncing data",
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/TargetDevices/BlazeTargetDevicePickerView.swift]: Super nit: Would a message like "Error loading devices. Please try again." be more helpful?

File: WooCommerce/Classes/ViewRelated/Blaze/TargetLanguages/BlazeTargetLanguagePickerView.swift [hunk 2]
```diff
@@ -50,11 +73,31 @@ private extension BlazeTargetLanguagePickerView {
...
+            comment: "Button to dismiss the target language picker for campaign creation"
+        )
+        static let saveButtonTitle = NSLocalizedString(
+            "blazeTargetLanguagePickerView.save",
+            value: "Save",
+            comment: "Button to save the selections on the target language picker for campaign creation"
+        )
+        static let errorMessage = NSLocalizedString(
+            "blazeTargetLanguagePickerView.errorMessage",
+            value: "Error syncing data",
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/TargetLanguages/BlazeTargetLanguagePickerView.swift]: Super nit: Same as the other comment. Kindly consider using a more descriptive error.

File: WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ErrorStateView.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+/// Reusable view for error state
+struct ErrorStateView: View {
+    let title: String
+    var subtitle: String?
+    var image: UIImage?
+    let actionTitle: String
+    let actionHandler: () -> Void
+
+    var body: some View {
+        VStack(spacing: 32) {
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ErrorStateView.swift]: nit: Consider adding a `Layout` enum to hold this spacing value.

File: WooCommerce/Classes/ViewRelated/Blaze/TargetLanguages/BlazeTargetLanguagePickerViewModel.swift [hunk 4]
```diff
@@ -91,6 +107,17 @@ private extension BlazeTargetLanguagePickerViewModel {
     /// and display the first fetch result immediately.
     ///
     func configureDisplayedData() {
+        $languages.combineLatest($isSyncingData, $syncError)
+            .map { devices, isSyncing, error -> SyncState in
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/TargetLanguages/BlazeTargetLanguagePickerViewModel.swift]: nit
```suggestion
            .map { languages, isSyncing, error -> SyncState in
```

**Review by selanthiraiyan — APPROVED** [review]
Thanks for working on this. This looks great. 🎉 I left a few non-blocking nits.

File: WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ErrorStateView.swift (hunk 3 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ErrorStateView.swift]: Done in ed0f1ef453d2b96b09728524d59a8e09bf98da48.

File: WooCommerce/Classes/ViewRelated/Blaze/TargetLanguages/BlazeTargetLanguagePickerViewModel.swift (hunk 4 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Blaze/TargetLanguages/BlazeTargetLanguagePickerViewModel.swift]: Fixed in 7edb33f7ff4fb93ed452d5889de73fbd6983b049.

File: WooCommerce/Classes/ViewRelated/Blaze/TargetDevices/BlazeTargetDevicePickerView.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Blaze/TargetDevices/BlazeTargetDevicePickerView.swift]: Makes sense! Updated in 1c678f697d3c5a67ba17b88d25dc57bb85810510.

File: WooCommerce/Classes/ViewRelated/Blaze/TargetLanguages/BlazeTargetLanguagePickerView.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Blaze/TargetLanguages/BlazeTargetLanguagePickerView.swift]: Also updated in 1c678f697d3c5a67ba17b88d25dc57bb85810510.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/180">16.9</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/180">16.9</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**